### PR TITLE
Convert metric creation from JSON to Map

### PR DESCRIPTION
**Pull request by arifBurakDemiray**

### CODE REVIEW

**Comment by ArtursKadikis** [comment]
waiting for @turtledreams

File: sdk/src/main/java/ly/count/android/sdk/DeviceInfo.java [hunk 1]
```diff
@@ -570,47 +570,48 @@ public String hasHinge(Context context) {
...
+            if (metricOverride.containsKey("_has_hinge")) {
+                map.put("_has_hinge", metricOverride.get("_has_hinge"));
             }
         }
 
-        return json;
+        return map;
+    }
+
+    private void putIfNotNullEndNotEmpty(@NonNull Map<String, Object> metrics, String key, String value) {
```
Comment (turtledreams) [inline sdk/src/main/java/ly/count/android/sdk/DeviceInfo.java:611]: possibly you meant 'putIfNotNull**A**ndNotEmpty'?

File: sdk/src/main/java/ly/count/android/sdk/DeviceInfo.java [hunk 2]
```diff
@@ -570,47 +570,48 @@ public String hasHinge(Context context) {
...
-                }
-                if (metricOverride.containsKey("_manufacturer")) {
-                    json.put("_manufacturer", metricOverride.get("_manufacturer"));
-                }
-                if (metricOverride.containsKey("_has_hinge")) {
-                    json.put("_has_hinge", metricOverride.get("_has_hinge"));
-                }
-            } catch (Exception ex) {
-                Countly.sharedInstance().L.e("[DeviceInfo] SDK encountered failure while trying to apply metric override, " + ex.toString());
+            if (metricOverride.containsKey("_device")) {
```
Comment (turtledreams) [inline sdk/src/main/java/ly/count/android/sdk/DeviceInfo.java:585]: You did not want to log sth here?

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: sdk/src/main/java/ly/count/android/sdk/DeviceInfo.java (hunk 2 above)
Comment (arifBurakDemiray) [reply sdk/src/main/java/ly/count/android/sdk/DeviceInfo.java:585]: added it for clarity

**Review by turtledreams — APPROVED** [review]

**Pull request merged**